webapp/views.py

```diff
@@ -226,23 +155,57 @@ def create_asset():
     friendly_name = flask.request.values.get("friendly-name")
     url_path = flask.request.values.get("url-path", "").strip("/")
 
+    data = {"tags": tags}
+    file_data = (b64decode(asset),)
+
+    if optimize:
+        try:
+            image = ImageProcessor(file_data)
+            image.optimize(allow_svg_errors=True)
+            file_data = image.data
+            data["optimized"] = True
+        except Exception:
+            # If optimisation failed, just don't bother optimising
+            data["optimized"] = False
+
+    if not url_path:
+        url_path = file_manager.generate_asset_path(file_data, friendly_name)
+
     try:
-        asset = _create_asset(
-            file_data=b64decode(asset),
-            friendly_name=friendly_name,
-            tags=tags,
-            url_path=url_path,
-            optimize=optimize,
-        )
-    except IOError as create_error:
-        if create_error.errno == errno.EEXIST:
-            return error_response(create_error)
-        else:
-            raise create_error
-    except SwiftException as swift_error:
-        return error_response(swift_error, url_path)
-
-    return flask.jsonify(asset), 201
+        with Image(blob=file_data) as image_info:
+            data["width"] = image_info.width
+            data["height"] = image_info.height
+    except Exception:
+        # Just don't worry if image reading fails
+        pass
+
+    asset = (
+        db_session.query(Asset)
+        .filter(Asset.file_path == url_path)
+        .one_or_none()
+    )
+
+    if asset:
+        if "width" not in asset.data and "width" in data:
+            asset.data["width"] = data["width"]
+
+        if "height" not in asset.data and "height" in data:
+            asset.data["height"] = data["height"]
+
+        db_session.commit()
+
+        error_message = "Asset already exists at {0}.".format(url_path)
```

Comment:
"Asset already exists at {url_path}."

Reply:
Changed here and in all other places where I was still using format.